torch/csrc/StorageSharing.cpp

```diff
@@ -312,36 +310,28 @@ static PyObject* THPStorage_shareCuda(PyObject* self, PyObject* noargs) {
   Py_INCREF(Py_None);
   THPObjectPtr _event_sync_required(Py_None);
   Py_INCREF(Py_None);
+
   if (storage.data()) {
-    // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-    auto shandle =
-        c10::cuda::CUDACachingAllocator::shareIpcHandle(storage.mutable_data());
-    _handle = PyBytes_FromStringAndSize(
-        shandle.handle.c_str(), (Py_ssize_t)shandle.handle.size());
-    _offset_bytes = PyLong_FromSsize_t((Py_ssize_t)shandle.offset);
-
-    // Put Storage Data behind new ref counting context
-    // See Note [CUDA IPC Refcounting implementation explained]
-    at::DataPtr sent_data_ptr = torch::GetNewRefCountedSentData(
-        storage.mutable_data(), storage.device());
-    auto old_data_ptr = storage.set_data_ptr(std::move(sent_data_ptr));
-    auto sent_data =
-        static_cast<torch::CudaIPCSentData*>(storage.data_ptr().get_context());
-    sent_data->set_original_ptr(std::move(old_data_ptr));
-    _ref_counter = PyBytes_FromString((sent_data->handle()).c_str());
-    _ref_counter_offset = THPUtils_packUInt64(sent_data->offset());
-
-    // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-    cudaIpcEventHandle_t ipc_event_handle;
-
-    if (sent_data->event_sync_required_) {
-      C10_CUDA_CHECK(
-          cudaIpcGetEventHandle(&ipc_event_handle, sent_data->event_));
-    }
+    auto ipc_tuple =
```

Comment:
nit: you can use structured binding to bind all the tuple elements in one go

Reply:
@ezyang Modified